packages/block/src/helpers.ts

```diff
@@ -158,7 +164,7 @@ export const getDataGasPrice = (header: BlockHeader) => {
  * @returns the total data gas fee for a transaction assuming it contains `numBlobs`
  */
 export const calcDataFee = (numBlobs: number, parent: BlockHeader) => {
-  if (parent.excessDataGas === undefined) {
+  if (parent.excessDataGas === undefined && parent._common.isActivatedEIP(4844)) {
```

Comment:
i think this is already validated in constructor of parent so not sure why we need extra validation here